src/mender-update/daemon/states.cpp

```diff
@@ -174,45 +193,21 @@ void SubmitInventoryState::DoSubmitInventory(Context &ctx, sm::EventPoster<State
 	}
 }
 
-void SubmitInventoryState::OnEnter(Context &ctx, sm::EventPoster<StateEvent> &poster) {
-	// Schedule timer for next update first, so that long running submissions do not postpone
-	// the schedule.
-	log::Debug(
-		"Scheduling the next inventory submission in: "
-		+ to_string(ctx.mender_context.GetConfig().inventory_poll_interval_seconds) + " seconds");
-	retry_.wait_timer.AsyncWait(
-		chrono::seconds(ctx.mender_context.GetConfig().inventory_poll_interval_seconds),
-		[&poster](error::Error err) {
-			if (err != error::NoError) {
-				if (err.code != make_error_condition(errc::operation_canceled)) {
-					log::Error("Inventory poll timer caused error: " + err.String());
-				}
-			} else {
-				poster.PostEvent(StateEvent::InventoryPollingTriggered);
-			}
-		});
-
-	DoSubmitInventory(ctx, poster);
-}
-
-PollForDeploymentState::PollForDeploymentState(
-	events::EventLoop &event_loop, int retry_interval_seconds, int retry_count) :
-	retry_ {
-		http::ExponentialBackoff(chrono::seconds(retry_interval_seconds), retry_count),
-		event_loop} {
+PollForDeploymentState::PollForDeploymentState(int retry_interval_seconds, int retry_count) :
+	backoff_ {chrono::seconds(retry_interval_seconds), retry_count} {
 }
 
 void PollForDeploymentState::HandlePollingError(Context &ctx, sm::EventPoster<StateEvent> &poster) {
```

Comment:
This `HandlePollingError` is still in two places, in here and in `SubmitInventoryState`.

We could move it by having the `backoff` objects also in the context. What do you think? Again the main disadvantage would be the logging, which in this case is info and error level so it is more relevant to get the more context for the (human) user reading the logs.

Reply:
> This `HandlePollingError` is still in two places, in here and in `SubmitInventoryState`.

I think that's fine. These two cases can potentially differ even more in the future.